overlord/devicestate/handlers.go

```diff
@@ -698,3 +703,231 @@ func fetchKeys(st *state.State, keyID string) (errAcctKey error, err error) {
 		keyID = a.SignKeyID()
 	}
 }
+
+func snapState(st *state.State, name string) (*snapstate.SnapState, error) {
```

Comment:
Nice, would love to see this as a separate change across the tree (just this function + conversion of all the hand-coded versions).

Reply:
I remember we had a discussion I long time ago about this and I had even a PR (thats totally outdated by now) but I don't remember the details :(